crate2nix/src/resolve.rs

```diff
@@ -616,9 +646,39 @@ impl CratesIoSource {
             version = self.version
         )
     }
+
+    /// Construct a URL for fetching a crate artifact based on the URL of the
+    /// registry index.
+    ///
+    /// For crates.io there is a known download URL we use:
+    ///   https://static.crates.io/crates/{name}/{name}-{version}.crate
+    ///
+    /// For custom registries we assume the index url has the form
+    ///   https://my-registry/path/to/index.git
+    /// And crate download URL has the form
+    ///   https://my-registry/path/to/{name}-{version}.crate
+    pub fn make_download_url(
+        name: String,
+        version: String,
+        index: &str,
+    ) -> Result<url::Url, Error> {
+        let mut download_url: String;
+        if index == "registry+https://github.com/rust-lang/crates.io-index" {
+            download_url = format!(
+                "https://static.crates.io/crates/{name}/{name}-{version}.crate",
+                name = name,
+                version = version
+            );
+        } else {
+            // Chop off the "registry+" prefix and "index.git" suffix
+            download_url = index[9..index.len() - 9].to_string();
```

Comment:
If really using the "prefix" trick, maybe chop off by its length rather than 9? Seems more readable